whois-nrtm4/src/main/java/net/ripe/db/nrtm4/util/NrtmFileUtil.java

```diff
@@ -0,0 +1,49 @@
+package net.ripe.db.nrtm4.util;
+
+import com.google.common.hash.Hashing;
+import net.ripe.db.nrtm4.publish.PublishableDeltaFile;
+import net.ripe.db.nrtm4.publish.PublishableSnapshotFile;
+import org.springframework.stereotype.Service;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Random;
+import java.util.UUID;
+
+import static net.ripe.db.nrtm4.NrtmConstants.DELTA_PREFIX;
+import static net.ripe.db.nrtm4.NrtmConstants.SNAPSHOT_PREFIX;
+
+
+@Service
+public class NrtmFileUtil {
+
+    private static final String CHARS = "0123456789abcdef";
+    private static final int RAND_STRING_LENGTH = 40;
+
+    public String fileName(final PublishableDeltaFile file) {
+        return String.format("%s.%d.%s.json", DELTA_PREFIX, file.getVersion(), randomHexString());
+    }
+
+    public String fileName(final PublishableSnapshotFile file) {
+        return String.format("%s.%d.%s.json", SNAPSHOT_PREFIX, file.getVersion(), randomHexString());
```

Comment:
These two function looks identical. Can we pass on the prefix from the deltaFileGenartor class (same goes for ssnapshot) ? If both snapshot and delta need to have same file format?

Reply:
we also need a 'notification' type which will not have a file name, since at any one point in time we only have one with a fixed name (no randomness). still, it could be that we decide to map the fixed name to the latest randomly generated one in a service. i'll change it.

Reply:
but also because i try to take advantage of the Java type system (polymorphism) instead of detecting types and using if...else to explicitly make decisions. it's usually less buggy that way. but still, i don't mind either way.